enterprise/ha/src/main/java/org/neo4j/kernel/ha/cluster/HighAvailabilityMemberStateMachine.java

```diff
@@ -235,16 +235,49 @@ public void notify( HighAvailabilityMemberListener listener )
             }
         }
 
+
+        /**
+         * As soon as we receive unavailability message and instanceId belongs us we depending from currentState
+         * will do the following:
+         * <lu>
+         * <li>if current state is <b>not</b> {@link HighAvailabilityMemberState#PENDING} we trigger switch to
+         * {@link
+         * HighAvailabilityMemberState#PENDING} and force new elections.</li>
+         * <li>if current state is {@link HighAvailabilityMemberState#PENDING}
+         * we only log debug message</li>
+         * </lu>
+         * The assumption here is: as soon as we receive unavailability event about us - then something went wrong
+         * in a cluster and we need to perform new elections.
+         * Elections should be triggered for all states except {@link HighAvailabilityMemberState#PENDING}, since
+         * first of all there is nothing or we already made a switch and waiting election to start, so no reason to
+         * start them again.
+         * <p>
+         * Listener invoked from sync block in {@link org.neo4j.cluster.member.paxos.PaxosClusterMemberEvents} so we
+         * should not have any racing here.
+         *</p>
+         * @param role The role for which the member is unavailable
+         * @param unavailableId The id of the member which became unavailable for that role
+         */
         @Override
         public void memberIsUnavailable( String role, InstanceId unavailableId )
         {
-            if ( context.getMyId().equals( unavailableId ) &&
-                 HighAvailabilityModeSwitcher.SLAVE.equals( role ) &&
-                 state == HighAvailabilityMemberState.SLAVE )
+            if ( context.getMyId().equals( unavailableId ) )
```

Comment:
We should have a test to make sure `memberIsUnavailable` moves to pending and calls for elections (or doesn't call elections).